components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/AccessTokenIssuer.java

```diff
@@ -707,6 +719,16 @@ private void addAuthorizedInternalScopes(OAuthTokenReqMessageContext tokReqMsgCt
         tokReqMsgCtx.setScope(scopesToReturn);
     }
 
+    private void addRequestedOIDCScopes(OAuthTokenReqMessageContext tokReqMsgCtx,
+                                             String[] requestedOIDCScopes) {
+        if (tokReqMsgCtx.getScope() == null) {
+            tokReqMsgCtx.setScope(new String[0]);
+        }
+        Set<String> scopesToReturn = new HashSet<>(Arrays.asList(tokReqMsgCtx.getScope()));
```

Comment:
Are we using a Set to remove duplicates?

Reply:
yes